barretenberg/cpp/src/barretenberg/commitment_schemes/shplonk/shplonk.hpp

```diff
@@ -97,48 +76,67 @@ template <typename Curve> class ShplonkProver_ {
      * @param z_challenge
      * @return Output{OpeningPair, Polynomial}
      */
-    static ShplonkProverOutput<Curve> compute_partially_evaluated_batched_quotient(
-        std::span<const OpeningPair<Curve>> opening_pairs,
-        std::span<const Polynomial> witness_polynomials,
-        Polynomial&& batched_quotient_Q,
+    static ProverOpeningClaim<Curve> compute_partially_evaluated_batched_quotient(
+        std::span<const ProverOpeningClaim<Curve>> opening_claims,
+        Polynomial& batched_quotient_Q,
         const Fr& nu_challenge,
         const Fr& z_challenge)
     {
-        const size_t num_opening_pairs = opening_pairs.size();
+        const size_t num_opening_claims = opening_claims.size();
 
         // {ẑⱼ(r)}ⱼ , where ẑⱼ(r) = 1/zⱼ(r) = 1/(r - xⱼ)
         std::vector<Fr> inverse_vanishing_evals;
-        inverse_vanishing_evals.reserve(num_opening_pairs);
-        for (const auto& pair : opening_pairs) {
-            inverse_vanishing_evals.emplace_back(z_challenge - pair.challenge);
+        inverse_vanishing_evals.reserve(num_opening_claims);
+        for (const auto& claim : opening_claims) {
+            inverse_vanishing_evals.emplace_back(z_challenge - claim.opening_pair.challenge);
         }
         Fr::batch_invert(inverse_vanishing_evals);
 
         // G(X) = Q(X) - Q_z(X) = Q(X) - ∑ⱼ ρʲ ⋅ ( fⱼ(X) − vⱼ) / ( r − xⱼ ),
         // s.t. G(r) = 0
-        Polynomial G(std::move(batched_quotient_Q)); // G(X) = Q(X)
+        Polynomial G(batched_quotient_Q); // G(X) = Q(X)
```

Comment:
Did you need to get rid of this move? It avoids making a full copy of this full sized polynomial doesn't it?

Reply:
I think you're right, here is a situation where std::move makes sense